### ISSUE TITLE
mAP^val results cannot be exactly reproduced

### ISSUE THREAD

**Issue by gyzou** [ultralytics/yolov5#4424]
Using pytorch 1.7.1 and tested on TitanV and TitanRTX separately, here are what I got on COCO val2017 with yolov5x and yolov5l:

yolov5l:
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95
                 all       5000      36335      0.728      0.615      0.663      0.469

yolov5x:
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 
                 all       5000      36335      0.749      0.626      0.682      0.491

**Comment by glenn-jocher** [comment]
@gyzou see Colab notebook:
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb

**Comment by gyzou** [comment]
> @gyzou see Colab notebook:
> https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb

That's right. My results are consistent with the numbers shown in the notebook. For example, using yolov5x.pt, we get 68.2 mAP@0.5. Not the 68.8 shown in the frontpage table

**Comment by glenn-jocher** [comment]
@gyzou I already told you, see the Colab notebook. I don't know how clearer I can be.

<img width="1053" alt="Screenshot 2021-08-16 at 10 48 28" src="https://user-images.githubusercontent.com/26833433/129537073-c4311c19-9f5e-4887-8d02-4b703ebefd38.png">

**Comment by zllljf** [comment]
When I trained with the parameters given on the official website:

**Comment by zllljf** [comment]
When I trained with the parameters given on the official website:
$ python train.py --data coco.yaml --cfg yolov5s.yaml --weights '' --batch-size 64
                                         yolov5m                                40
                                         yolov5l                                24
                                         yolov5x                                16

**Comment by zllljf** [comment]
> @gyzou I already told you, see the Colab notebook. I don't know how clearer I can be.
> 
> <img alt="Screenshot 2021-08-16 at 10 48 28" width="1053" src="https://user-images.githubusercontent.com/26833433/129537073-c4311c19-9f5e-4887-8d02-4b703ebefd38.png">

I can not got related iou with the way of eval like:
`python val.py --weights last.pt --data coco.yaml --img 640 --iou 0.65 --half`
when I trained with the parameters given on the official website:
```
$ python train.py --data coco.yaml --cfg yolov5s.yaml --weights '' --batch-size 64
                                                                 yolov5m                                40
                                                                 yolov5l                                  24
                                                                 yolov5x                                 16
```
 here are what I got on COCO val2017 with last.pt saved in runs/train/exp/weights after train:
 yolov5l:
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.467

yolov5m:
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.437

Whether the yolov5l.pt yolov5m.pt and yolov5x.pt given on the official website trained by diff parameters?

**Comment by zllljf** [comment]
@gyzou @glenn-jocher 
The model trained according to the training parameters given on the official website cannot get the corresponding IOU when eval.
But eval with *pt given on the official website can get the corresponding IOU
Is the model given on the official website trained with parameters are different with parameters given on the official website? 
Here are code what i trained:
```
 python train.py --data coco.yaml --cfg yolov5m.yaml --weights '' --batch-size 40
 python train.py --data coco.yaml --cfg yolov5l.yaml --weights '' --batch-size 24
```

**Comment by zllljf** [comment]
@gyzou @glenn-jocher 
The model trained according to the training parameters given on the official website cannot get the corresponding IOU when eval.
But eval with *pt given on the official website can get the corresponding IOU
Is the model given on the official website trained with parameters are different with parameters given on the official website? 
Here are code what i trained:
```
 python train.py --data coco.yaml --cfg yolov5m.yaml --weights '' --batch-size 40
 python train.py --data coco.yaml --cfg yolov5l.yaml --weights '' --batch-size 24
```

**Comment by glenn-jocher** [comment]
@zllljf ah sorry I didn't understand. The models in the v5.0 release were trained with the P6 hyperparameters, which favor large models vs small models.
```
python train.py --hyp hyp.scratch-p6.yaml
```

https://github.com/ultralytics/yolov5/blob/dbc06ce29298de4a55d73fa37362dd51a03035ed/data/hyps/hyp.scratch-p6.yaml#L2-L34

**Comment by zllljf** [comment]
@glenn-jocher en? not hypscratch-p6.yaml but hyp.yaml
when i train yolov5m or yolov5l, i train with the paras like this:
![image](https://user-images.githubusercontent.com/25974707/129564934-e4dcf934-3b27-4a25-97f9-3bb1cfbed876.png)
and follow the instructions ， i will get the IOU like this:
![image](https://user-images.githubusercontent.com/25974707/129564745-f9170019-bc6c-49f7-bb59-85e8a311caf2.png)
but i can not get corresponding IOU
so I wonder if the parameters are wrong？

**Comment by zllljf** [comment]
> @zllljf ah sorry I didn't understand. The models in the v5.0 release were trained with the P6 hyperparameters, which favor large models vs small models.
> 
> ```
> python train.py --hyp hyp.scratch-p6.yaml
> ```
> 
> https://github.com/ultralytics/yolov5/blob/dbc06ce29298de4a55d73fa37362dd51a03035ed/data/hyps/hyp.scratch-p6.yaml#L2-L34

In short , I git clone the source code, and trained according to the parameters given in the tutorial like this :
```
 python train.py --data coco.yaml --cfg yolov5m.yaml --weights '' --batch-size 40
 python train.py --data coco.yaml --cfg yolov5l.yaml --weights '' --batch-size 24
```

and eval the saved last.pt, can not get the corresponding mAP given in the tutorial

**Comment by glenn-jocher** [comment]
@zllljf 
```
python train.py --data coco.yaml --cfg yolov5l.yaml --weights '' --batch-size 24 --hyp hyp.scratch-p6.yaml
```

See argparser for a full list of arguments.
